src/internal/maper/maper.go

```diff
@@ -0,0 +1,136 @@
+package maper
+
+import (
+	"github.com/futurehomeno/cliffhanger/adapter/service/chargepoint"
+	log "github.com/sirupsen/logrus"
+)
+
+func SupportedPhaseModes(gridType chargepoint.GridType, phaseMode int, phases int) []chargepoint.PhaseMode {
+	gridTypeMap, ok := phaseModeMatrix[gridType]
+	if !ok {
+		log.Errorf("can't set supported phase modes for gridType: %v", gridType)
+
+		return []chargepoint.PhaseMode{}
+	}
+
+	phasesMap, ok := gridTypeMap[phases]
+	if !ok {
+		log.Errorf("can't set supported phase modes for phases: %v", phases)
+
+		return []chargepoint.PhaseMode{}
+	}
+
+	phaseModeMap, ok := phasesMap[phaseMode]
+	if !ok {
+		log.Errorf("can't set supported phase modes for easee phase mode: %v", phaseMode)
+
+		return []chargepoint.PhaseMode{}
+	}
+
+	return phaseModeMap
+}
+
+var phaseModeMatrix = map[chargepoint.GridType]map[int]map[int][]chargepoint.PhaseMode{
+	chargepoint.GridTypeTN: {
+		1: {
+			1: []chargepoint.PhaseMode{chargepoint.PhaseModeNL1},
+			2: []chargepoint.PhaseMode{chargepoint.PhaseModeNL1},
+		},
+		3: {
+			1: []chargepoint.PhaseMode{chargepoint.PhaseModeNL1, chargepoint.PhaseModeNL2, chargepoint.PhaseModeNL3},
+			2: []chargepoint.PhaseMode{chargepoint.PhaseModeNL1, chargepoint.PhaseModeNL2, chargepoint.PhaseModeNL3, chargepoint.PhaseModeNL1L2L3},
+			3: []chargepoint.PhaseMode{chargepoint.PhaseModeNL1L2L3},
+		},
+	},
+	chargepoint.GridTypeTT: {
+		1: {
+			1: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2},
+			2: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2},
+		},
+		3: {
+			1: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2, chargepoint.PhaseModeL2L3, chargepoint.PhaseModeL3L1},
+			2: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2, chargepoint.PhaseModeL2L3, chargepoint.PhaseModeL3L1, chargepoint.PhaseModeL1L2L3},
+			3: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2L3},
+		},
+	},
+	chargepoint.GridTypeIT: {
+		1: {
+			1: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2},
+			2: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2},
+		},
+		3: {
+			1: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2, chargepoint.PhaseModeL2L3, chargepoint.PhaseModeL3L1},
+			2: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2, chargepoint.PhaseModeL2L3, chargepoint.PhaseModeL3L1, chargepoint.PhaseModeL1L2L3},
+			3: []chargepoint.PhaseMode{chargepoint.PhaseModeL1L2L3},
+		},
+	},
+}
+
+// GridType represents a grid type.
+type GridType int
```

Comment:
Let's create a `model` package, where we'll be storing all the models used across the application. In general I favor domain-oriented packages (e.g. `http`, `signalr`, `easee`) versus function-oriented (`repository`, `model`, `service`), but in this scenario it will be much more useful than juggling stucts and other types back and forth between different packages, just to satisfy the compiler.